src/client/components/app/Transactions/DepositTx.tsx

```diff
@@ -66,16 +68,23 @@ export const DepositTx: FC<DepositTxProps> = ({
   const expectedTxOutcome = useAppSelector(VaultsSelectors.selectExpectedTxOutcome);
   const expectedTxOutcomeStatus = useAppSelector(VaultsSelectors.selectExpectedTxOutcomeStatus);
   const actionsStatus = useAppSelector(VaultsSelectors.selectSelectedVaultActionsStatusMap);
+  const selectedTokenMap = useAppSelector(TokensSelectors.selectTokensMap);
 
   const sellTokensOptions = selectedVault
     ? [selectedVault.token, ...userTokens.filter(({ address }) => address !== selectedVault.token.address)]
     : userTokens;
   const sellTokensOptionsMap = keyBy(sellTokensOptions, 'address');
-  const selectedSellToken = sellTokensOptionsMap[selectedSellTokenAddress ?? ''];
+  let selectedSellToken: TokenView | undefined = sellTokensOptionsMap[selectedSellTokenAddress ?? ''];
   const [tokenAllowance, isLoadingTokenAllowance, tokenAllowanceError] = useAllowance({
     tokenAddress: selectedSellTokenAddress,
     vaultAddress: selectedVault?.address,
   });
+  const addressIsNativeCurrency = selectedSellTokenAddress === ETHEREUM_ADDRESS;
+  const shouldUseNativeCurrency = typeof selectedSellToken === 'undefined' && addressIsNativeCurrency;
+  if (shouldUseNativeCurrency) {
+    const tokenData = selectedTokenMap[ETHEREUM_ADDRESS];
+    selectedSellToken = createPlaceholderToken(tokenData);
+  }
```

Comment:
I feel this component is doing way too much, what do you think @huggingbot if we refactor this to logic to a function so that we can do something like 

```typescript
const selectedSellToken = getSelectedSellToken({ ... });
```

I think we could avoid doing `typeof selectedSellToken === 'undefined'`, I think all we care is if there's a `selectedSellToken` or not, if there isn't one and addressIsNativeCurrency then return the `createPlaceholderToken(tokenData)`, otherwise we go with the result from the map

Reply:
It's looking better IMO, decluttering the main logic. Small suggestion in https://github.com/yearn/yearn-finance-v3/pull/584#discussion_r851561518